src/opnsense/scripts/unbound/start.sh

```diff
@@ -26,11 +26,24 @@
 # OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 # SUCH DAMAGE.
 
-set -e
-
 # prepare and startup unbound, so we can easily background it
 
-chroot -u unbound -g unbound / /usr/local/sbin/unbound-anchor -a /var/unbound/root.key
+# if the root.key file is missing or damaged, run unbound-anchor
+if ! /usr/local/sbin/unbound-checkconf /var/unbound/unbound.conf 2> /dev/null; then
```

Comment:
now that this is guarded by "if !" the set -e could remain where it originally was. if we really worry about unbound-anchor returning errors we may as well use "if !" with it to emit a log message in that case or use "|| true" and move the error return comment above the invoke where it belongs. it's easy to miss this otherwise later on

Reply:
unbound-anchor will return non-zero on failover, which is still valid for bootstrapping.